### PR TITLE
Adding support to panelAR objects ("panelAR" package)

### PR DESCRIPTION
**Pull request by alexeygridnev**
Resolving #107 issue - adding support for panelAR models (panelAR) package.

### CODE REVIEW

**Comment by czucca** [comment]
Hi Alexey, thanks for the PR. I've spotted a couple of errors in your code:

- in include nobs you are assigning rs to gofs for the second time instead of nobs, 
- when you set the method instead of stats is "panelAR" (tha name of the package)
- make sure that panelAR is the class of the regression output and not the function you use to run the regression (sorry if you already did this, I'm just trying to help)
-also, could you please use arrows instead of =

many Thanks for your contribution, 
C.

**Comment by alexeygridnev** [comment]
Hello! Thank you for the comments!
I fixed all the issues you mentioned, however, Travis CI is still showing
errors. I will try to have a look again what's wrong.

Best regards,
Aleksei Gridnev

**Comment by czucca** [comment]
As far as I'm seeing this, you have forgotten to update extract.Rd. If you don't do so it won't compile. 
Also, there is still one = when you assign the function. Can you please replace with <-?

Thanks, 
C.

**Comment by alexeygridnev** [comment]
Thanks! Now the checks are passed.

**Comment by leifeld** [comment]
@czucca or @alexeygridnev Can you please resolve the conflicts?
@czucca Can you give this an initial review? Once you approve,I'll review it asap and merge it.
Thanks both!

File: R/extract.R [hunk 1]
```diff
@@ -5147,6 +5147,47 @@ extract.pglm <- function(model, include.aic = TRUE,
...
     return(tr)
 }
-
 setMethod("extract", signature = className("pglm", "pglm"),
           definition = extract.pglm)
+					 
+					 
+#extention for panelAR objects (panelAR package)					 
+extract.panelAR <- function(model, include.rsquared=TRUE, include.nobs=TRUE, include.groups=TRUE,...){
+    s <- summary(model,...)
```
Comment (czucca) [inline R/extract.R]: Can you please leave a blank line?

File: R/extract.R [hunk 2]
```diff
@@ -5147,6 +5147,47 @@ extract.pglm <- function(model, include.aic = TRUE,
...
+    names <- rownames(s$coef)
+    co <- s$coef[, 1]
+    se <- s$coef[, 2]
+    pval <- s$coef[, 4]
```
Comment (czucca) [inline R/extract.R]: Can you please leave a blank line?

File: vignettes/texreg.Rnw [hunk 3]
```diff
@@ -504,6 +504,7 @@ Class			& Package			& Description \\
 `\code{netlogit}'		& \pkg{sna}			& Logistic regression for networks \\
 `\code{nlme}'			& \pkg{nlme}			& Nonlinear mixed-effects models \\
 `\code{ols}'			& \pkg{rms}			& Ordinary least squares linear models \\
+`\code{panelAR}'			& \pkg{panelAR}			& AR(1)-autocorrelation and panel heteroskedasticity/contemporaneous correlation\\
```
Comment (czucca) [inline vignettes/texreg.Rnw]: can you please replace the note with: "Linear models on panel data structures in the presence of AR(1)-type autocorrelation" ?

File: man/extract.Rd [hunk 4]
```diff
@@ -135,6 +135,8 @@
 \alias{extract,nlmerMod-method}
 \alias{extract.ols}
 \alias{extract,ols-method}
+\alias{extract.panelAR}
+\alias{extract,panelAR-method}
 \alias{extract.pglm}
 \alias{extract,pglm-method}
```
Comment (czucca) [inline man/extract.Rd]: can you please insert below in the appropriate line (alphabetic order): 

 \S4method{extract}{panelAR}(model, include.rsquared = TRUE,  include.nobs = TRUE,
  include.group = TRUE, ...)

File: R/extract.R [hunk 5]
```diff
@@ -5147,6 +5147,47 @@ extract.pglm <- function(model, include.aic = TRUE,
...
 }
-
 setMethod("extract", signature = className("pglm", "pglm"),
           definition = extract.pglm)
+					 
+					 
+#extention for panelAR objects (panelAR package)					 
+extract.panelAR <- function(model, include.rsquared=TRUE, include.nobs=TRUE, include.groups=TRUE,...){
+    s <- summary(model,...)
+    names <- rownames(s$coef)
```
Comment (czucca) [inline R/extract.R]: can you please replace "names" with "coefficient.names"? --names is an inbuilt R function and might generate conflicts.

File: R/extract.R [hunk 6]
```diff
@@ -5147,6 +5147,47 @@ extract.pglm <- function(model, include.aic = TRUE,
...
+    	gof.decimal <- c(gof.decimal, TRUE)
+    }	
+    if (include.groups == TRUE){
+	ngroups <- sqrt(length(s$Sigma))
+    	gof <- c(gof, ngroups)
+    	gof.names <- c(gof.names, "Num. \\ panels")
+    	gof.decimal <- c(gof.decimal, FALSE)
+    }
+   
+    tr <- createTexreg(coef.names = names,
```
Comment (czucca) [inline R/extract.R]: coefficient.names here too please

**Comment by czucca** [comment]
Hi Aleksey, 

can you please review my comments? don't worry about the conflicts, I'll take care of them. 

Thanks, 
Claudia

**Comment by alexeygridnev** [comment]
Done!

**Comment by leifeld** [comment]
@czucca Where do we stand with this one? Can I review, or are there any pending changes?

**Comment by czucca** [comment]
yes please review, I'm done with this one!

File: NAMESPACE [hunk 7]
```diff
@@ -28,7 +28,6 @@ export("extract.bam")
 export("extract.betamfx")
 export("extract.betaor")
 export("extract.betareg")
-export("extract.bife")
```
Comment (leifeld) [inline NAMESPACE]: Why is this deleted?

File: NAMESPACE [hunk 8]
```diff
@@ -45,7 +44,6 @@ export("extract.dynlm")
 export("extract.ergm")
 export("extract.ergmm")
 export("extract.ets")
-export("extract.feglm")
```
Comment (leifeld) [inline NAMESPACE]: Why is this deleted?

File: NAMESPACE [hunk 9]
```diff
@@ -59,7 +57,6 @@ export("extract.glmmadmb")
 export("extract.glmmPQL")
 export("extract.glmrob")
 export("extract.gls")
-export("extract.gnls")
```
Comment (leifeld) [inline NAMESPACE]: Why is this deleted?

File: NAMESPACE [hunk 10]
```diff
@@ -78,7 +75,6 @@ export("extract.lqmm")
 export("extract.lrm")
 export("extract.maBina")
 export("extract.mer")
-export("extract.mhurdle")
```
Comment (leifeld) [inline NAMESPACE]: Why is this deleted?

File: NAMESPACE [hunk 11]
```diff
@@ -90,8 +86,8 @@ export("extract.negbin")
 export("extract.netlogit")
 export("extract.nlme")
 export("extract.nlmerMod")
-export("extract.oglmx")
```
Comment (leifeld) [inline NAMESPACE]: Why is this deleted?

File: R/extract.R [hunk 12]
```diff
@@ -5441,4 +5441,47 @@ extract.gnm <- function(model, include.aic = TRUE, include.bic = TRUE,
 }
 
 setMethod("extract", signature = className("gnm", "gnm"),
-          definition = extract.gnm)
\ No newline at end of file
+          definition = extract.gnm)
```
Comment (leifeld) [inline R/extract.R]: Insert blank line before the comment.

File: R/extract.R [hunk 13]
```diff
@@ -5441,4 +5441,47 @@ extract.gnm <- function(model, include.aic = TRUE, include.bic = TRUE,
...
+# extention for panelAR objects (panelAR package)					 
+extract.panelAR <- function(model, include.rsquared=TRUE, include.nobs=TRUE, include.groups=TRUE,...){
```
Comment (leifeld) [inline R/extract.R]: Add spaces around `=`.

File: R/extract.R [hunk 14]
```diff
@@ -5441,4 +5441,47 @@ extract.gnm <- function(model, include.aic = TRUE, include.bic = TRUE,
...
+        if (include.nobs == TRUE){
+        nobs <- length(s$residuals)
+    	gof <- c(gof, nobs)
+    	gof.names <- c(gof.names, "Num.\\ obs.")
+    	gof.decimal <- c(gof.decimal, TRUE)
+    }	
+    if (include.groups == TRUE){
+	ngroups <- sqrt(length(s$Sigma))
+    	gof <- c(gof, ngroups)
+    	gof.names <- c(gof.names, "Num. \\ panels")
```
Comment (leifeld) [inline R/extract.R]: Remove space before `\\`.

File: man/extract.Rd [hunk 15]
```diff
@@ -518,6 +520,9 @@ extract(model, ...)
 \S4method{extract}{ols}(model, include.nobs = TRUE,
     include.rsquared = TRUE, include.adjrs = TRUE,
     include.fstatistic = FALSE, include.lr = TRUE, ...)
+  
+ \S4method{extract}{panelAR}(model, include.rsquared=TRUE, 
+    include.nobs=TRUE, include.groups=TRUE,...)
```
Comment (leifeld) [inline man/extract.Rd]: Add spaces around `=`.

File: vignettes/texreg.Rnw [hunk 16]
```diff
@@ -510,6 +510,7 @@ Class			& Package			& Description \\
 `\code{nlme}'           & \pkg{nlme}            & Nonlinear mixed-effects models \\
 `\code{oglmx}'          & \pkg{oglmx}           & Ordered Generalized Linear Model \\
 `\code{ols}'            & \pkg{rms}             & Ordinary least squares linear models \\
+`\code{panelAR}'        & \pkg{panelAR}         & Linear models on panel data structures in the presence of AR(1)-type autocorrelation \\
```
Comment (leifeld) [inline vignettes/texreg.Rnw]: Increase the model count above the table?

**Comment by leifeld** [comment]
Great, thanks! I'll close this one and we'll take it from there!

Thanks again, @alexeygridnev, for the input!
